flink-queryable-state/flink-queryable-state-runtime/src/test/java/org/apache/flink/queryablestate/itcases/NonHAQueryableStateFsBackendITCase.java

```diff
@@ -48,31 +48,33 @@ public class NonHAQueryableStateFsBackendITCase extends AbstractQueryableStateTe
     private static final int QS_PROXY_PORT_RANGE_START = 9084;
     private static final int QS_SERVER_PORT_RANGE_START = 9089;
 
-    @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
-
-    @ClassRule
-    public static final MiniClusterWithClientResource MINI_CLUSTER_RESOURCE =
```

Comment:
Can it be replaced by `MiniClusterExtension`?

Reply:
For `HAQueryableStateFsBackendITCase` and `HAQueryableStateRocksDBBackendITCase`, the initialization of `MiniClusterExtension` rely on `@Tempdir`, which will cause NPE,  I don't know how to initialize `@Tempdir` before `@RegisterExtension`,  do you have any suggestion?

Reply:
Maybe we can use `Order` annotation provided by `Junit5`, for example:

```
class HAQueryableStateFsBackendITCase extends AbstractQueryableStateTestBase {

    private static final int NUM_JMS = 2;
    // NUM_TMS * NUM_SLOTS_PER_TM must match the parallelism of the pipelines so that
    // we always use all TaskManagers so that the JM oracle is always properly re-registered
    private static final int NUM_TMS = 2;
    private static final int NUM_SLOTS_PER_TM = 2;

    private static final int QS_PROXY_PORT_RANGE_START = 9064;
    private static final int QS_SERVER_PORT_RANGE_START = 9069;

    @TempDir
    @Order(1)
    public static Path tmpStateBackendDir;

    @TempDir
    @Order(2)
    public static Path tmpHaStoragePath;

    @RegisterExtension
    @Order(3)
    public static final AllCallbackWrapper<ZooKeeperExtension> ZK_RESOURCE =
            new AllCallbackWrapper<>(new ZooKeeperExtension());

    @RegisterExtension
    @Order(4)
    public static final MiniClusterExtension MINI_CLUSTER_RESOURCE =
            new MiniClusterExtension(
                    () ->
                            new MiniClusterResourceConfiguration.Builder()
                                    .setConfiguration(getConfig())
                                    .setNumberTaskManagers(NUM_TMS)
                                    .setNumberSlotsPerTaskManager(NUM_SLOTS_PER_TM)
                                    .build());

    @Override
    protected StateBackend createStateBackend() throws Exception {
        return new FsStateBackend(tmpStateBackendDir.toUri().toString());
    }

    @BeforeAll
    static void setup(@InjectClusterClient RestClusterClient<?> injectedClusterClient)
            throws Exception {
        client = new QueryableStateClient("localhost", QS_PROXY_PORT_RANGE_START);

        clusterClient = injectedClusterClient;
    }

    @AfterAll
    static void tearDown() throws Exception {
        client.shutdownAndWait();
    }

    private static Configuration getConfig() {

        Configuration config = new Configuration();
        config.setBoolean(QueryableStateOptions.ENABLE_QUERYABLE_STATE_PROXY_SERVER, true);
        config.set(TaskManagerOptions.MANAGED_MEMORY_SIZE, MemorySize.parse("4m"));
        config.setInteger(ConfigConstants.LOCAL_NUMBER_JOB_MANAGER, NUM_JMS);
        config.setInteger(ConfigConstants.LOCAL_NUMBER_TASK_MANAGER, NUM_TMS);
        config.setInteger(TaskManagerOptions.NUM_TASK_SLOTS, NUM_SLOTS_PER_TM);
        config.setInteger(QueryableStateOptions.CLIENT_NETWORK_THREADS, 2);
        config.setInteger(QueryableStateOptions.PROXY_NETWORK_THREADS, 2);
        config.setInteger(QueryableStateOptions.SERVER_NETWORK_THREADS, 2);
        config.setString(
                QueryableStateOptions.PROXY_PORT_RANGE,
                QS_PROXY_PORT_RANGE_START + "-" + (QS_PROXY_PORT_RANGE_START + NUM_TMS));
        config.setString(
                QueryableStateOptions.SERVER_PORT_RANGE,
                QS_SERVER_PORT_RANGE_START + "-" + (QS_SERVER_PORT_RANGE_START + NUM_TMS));
        config.setBoolean(WebOptions.SUBMIT_ENABLE, false);

        config.setString(HighAvailabilityOptions.HA_STORAGE_PATH, tmpHaStoragePath.toString());

        config.setString(
                HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM,
                ZK_RESOURCE.getCustomExtension().getConnectString());
        config.setString(HighAvailabilityOptions.HA_MODE, "zookeeper");

        return config;
    }
}
```

Reply:
👍Thanks for the guidance, I replaced all relevant `MiniClusterWithClientResource` with `MiniClusterExtension`.